src/main/java/seedu/address/logic/commands/patient/SearchPatientCommand.java

```diff
@@ -79,14 +90,24 @@ public CommandResult execute(Model model) throws CommandException {
     public CommandResult findPatientWithName(SearchPatientDescriptor searchPatientDescriptor,
                                              List<Patient> patientList) throws CommandException {
         String nameToSearch = searchPatientDescriptor.getName().toString().trim().toLowerCase();
+        ArrayList<Patient> patientNameList = new ArrayList<>();
 
         for (Patient patient : patientList) {
             String patientName = patient.getName().toString().trim().toLowerCase();
-            if (patientName.equals(nameToSearch)) {
-                return new CommandResult(String.format(MESSAGE_SEARCH_PATIENT_SUCCESS, patient));
+            if (patientName.contains(nameToSearch)) {
+                patientNameList.add(patient);
             }
         }
-        throw new CommandException(MESSAGE_PATIENT_NOT_FOUND);
+
+        if (patientNameList.isEmpty()) {
+            throw new CommandException(MESSAGE_PATIENT_NOT_FOUND);
+        } else if (patientNameList.size() == 1) {
+            return new CommandResult(String.format(MESSAGE_SEARCH_PATIENT_SUCCESS,
+                    patientNameList.get(0)));
+        } else {
```

Comment:
Can you just combine the two methods, seems like not much of a difference.

Reply:
yup! is there a reason why you are checking if the search result has only has one value? I'm referencing to the ``(patientNameList.size() == 1)`` cuz your ``getListOutput(list)`` method will just print out everything

Reply:
The main reason I put this is just because I feel that it is not necessary to output a list when there is only one item